code/game/machinery/vending/new_marine_vendors.dm

```diff
@@ -49,6 +49,14 @@
 
 	var/list/listed_products
 
+/obj/machinery/marine_selector/power_change()
+	..()
+	if( !(stat & NOPOWER) )
+		icon_state = initial(icon_state)
+	else
+		spawn(rand(0, 15))
+			src.icon_state = "[initial(icon_state)]-off"
```

Comment:
Why a spawn?
Also `src.` is not needed

Reply:
Basically this is copied from vending template, I suppose the `spawn` was added to make the row of vendors in the preps not go off in unison, an aestethic choice from the days where the marine vendors were a subtype of vending. But yes, I'll remove the `src.` from both .dm.